lib/kafka/async_producer.rb

```diff
@@ -141,11 +143,17 @@ def shutdown
     private
 
     def ensure_threads_running!
+      raise ShuttingDownError if @shutting_down
```

Comment:
What behavior do we want here? I think that `@shutting_down` should get set in `shutdown`, but then maybe only raise `ShuttingDownError` if any of the threads are dead?

Reply:
Do you have a call stack when the shutdown problem happens? Maybe it would be enough to just `return` without starting the threads.